core/src/main/java/com/sequenceiq/cloudbreak/rotation/saltboot/SaltBootRotationExecutor.java

```diff
@@ -0,0 +1,272 @@
+        if (!isRotatedSecrets(saltBootPassword, saltBootPrivateKey)) {
+            throw new SecretRotationException("Salt boot password and private key is not rotated.", getType());
+        } else {
+            try {
+                updateSaltBootPublicKey(stack, saltBootPrivateKey, RotationSecret::getSecret);
+                GatewayConfig oldPrimaryGateway = withOldSecrets(gatewayConfigService.getPrimaryGatewayConfig(stack), saltBootPassword, saltBootPrivateKey);
+                if (isSaltBootReachableWithGatewayConfig(stack, oldPrimaryGateway)) {
+                    uploadNewSaltBootConfig(stack, oldPrimaryGateway, saltBootPassword.getSecret(), saltBootPrivateKey.getSecret());
+                    restartSaltBootService(stack, oldPrimaryGateway);
+                } else {
+                    LOGGER.info("Salt boot is already configured with the rotated keys.");
+                }
+            } catch (Exception e) {
+                LOGGER.error("Error happened during salt boot secret rotation.", e);
+                throw new SecretRotationException("Exception during salt boot secret rotation.", e, getType());
+            }
+        }
+    }
+
+    private void updateSaltBootPublicKey(StackDto stack, RotationSecret saltBootPrivateKey, Function<RotationSecret, String> mapper) {
+        SaltSecurityConfig saltSecurityConfig = stack.getSecurityConfig().getSaltSecurityConfig();
+        saltSecurityConfig.setSaltBootSignPublicKey(calcSaltBootPublicKey(mapper.apply(saltBootPrivateKey)));
+        securityConfigService.save(saltSecurityConfig);
+    }
+
+    private String calcSaltBootPublicKey(String privateKey) {
+        PublicKey publicKey = PkiUtil.getPublicKey(new String(BASE64.decode(privateKey)));
+        String openSshFormatPublicKey = PkiUtil.convertOpenSshPublicKey(publicKey);
+        return BASE64.encode(openSshFormatPublicKey.getBytes());
+    }
+
+    @Override
+    public void rollback(SaltBootRotationContext rotationContext) {
+        StackDto stack = stackService.getByCrn(rotationContext.getResourceCrn());
+        RotationSecret saltBootPassword = secretService.getRotation(rotationContext.getSaltBootPasswordSecretPath());
+        RotationSecret saltBootPrivateKey = secretService.getRotation(rotationContext.getSaltBootPrivateKeySecretPath());
+        GatewayConfig primaryGateway = gatewayConfigService.getPrimaryGatewayConfig(stack);
+        if (!isRotatedSecrets(saltBootPassword, saltBootPrivateKey)) {
+            LOGGER.info("Checking if salt boot is reachable with current secrets.");
+            if (!isSaltBootReachableWithGatewayConfig(stack, primaryGateway)) {
+                String msg = "Salt boot is not reachable with the current password and private key and there are no backup values present. " +
+                        "This might happened because of an issue in previous salt boot secret.";
+                LOGGER.error(msg);
+                throw new SecretRotationException(msg, getType());
```

Comment:
right now this is ok, later we can move this into prevalidation phase of the framework when it is implemented